primitives/types/metadata_generator.go

```diff
@@ -0,0 +1,237 @@
+package types
+
+import (
+	"reflect"
+	"strings"
+
+	sc "github.com/LimeChain/goscale"
+	"github.com/LimeChain/gosemble/constants/metadata"
+)
+
+const (
+	additionalSignedTypeName = "typesInfoAdditionalSignedData"
+	moduleTypeName           = "Module"
+	hookOnChargeTypeName     = "OnChargeTransaction"
+	varyingDataTypeName      = "VaryingData"
+)
+
+type MetadataTypeGenerator struct {
+	MetadataIds   map[string]int
+	MetadataTypes sc.Sequence[MetadataType]
+}
+
+func NewMetadataTypeGenerator() MetadataTypeGenerator {
+	return MetadataTypeGenerator{
+		MetadataIds:   BuildMetadataTypesIdsMap(),
+		MetadataTypes: sc.Sequence[MetadataType]{},
+	}
+}
+
+func BuildMetadataTypesIdsMap() map[string]int {
+	return map[string]int{
+		"Bool":                       metadata.PrimitiveTypesBool,
+		"String":                     metadata.PrimitiveTypesString,
+		"U8":                         metadata.PrimitiveTypesU8,
+		"U16":                        metadata.PrimitiveTypesU16,
+		"U32":                        metadata.PrimitiveTypesU32,
+		"U64":                        metadata.PrimitiveTypesU64,
+		"U128":                       metadata.PrimitiveTypesU128,
+		"I8":                         metadata.PrimitiveTypesI8,
+		"I16":                        metadata.PrimitiveTypesI16,
+		"I32":                        metadata.PrimitiveTypesI32,
+		"I64":                        metadata.PrimitiveTypesI64,
+		"I128":                       metadata.PrimitiveTypesI128,
+		"H256":                       metadata.TypesH256,
+		"SequenceU8":                 metadata.TypesSequenceU8,
+		"MultiAddress":               metadata.TypesMultiAddress,
+		"Header":                     metadata.Header,
+		"SequenceUncheckedExtrinsic": metadata.TypesSequenceUncheckedExtrinsics,
+	}
+}
+
+func (g *MetadataTypeGenerator) IdsMap() map[string]int {
+	return g.MetadataIds
+}
+
+func (g *MetadataTypeGenerator) GetMetadataTypes() sc.Sequence[MetadataType] {
+	return g.MetadataTypes
+}
```

Comment:
nit: either all or none of the getter fns should have Get prefix for consistency